lms/djangoapps/oauth_dispatch/tests/test_views.py

```diff
@@ -153,43 +153,78 @@ class TestAuthorizationView(TestCase):
+        self.dot_app = self.dot_adapter.create_confidential_client(
+            name='test dot application',
+            user=self.user,
+            redirect_uri=DUMMY_REDIRECT_URL,
+            client_id='confidential-dot-app-client-id',
+        )
+        self.dop_app = self.dop_adapter.create_confidential_client(
+            name='test dop client',
+            user=self.user,
+            redirect_uri=DUMMY_REDIRECT_URL,
+            client_id='confidential-dop-app-client-id',
         )
 
-    def test_authorization_view(self):
+    @ddt.data('dop', 'dot')
+    def test_post_authorization_view(self, client_type):
+        oauth_application = getattr(self, '{}_app'.format(client_type))
         self.client.login(username=self.user.username, password='test')
+        allow_field = {'dop': 'authorize', 'dot': 'allow'}
         response = self.client.post(
             '/oauth2/authorize/',
             {
-                'client_id': self.dop_client.client_id,  # TODO: DOT is not yet supported (MA-2124)
+                'client_id': oauth_application.client_id,
                 'response_type': 'code',
                 'state': 'random_state_string',
                 'redirect_uri': DUMMY_REDIRECT_URL,
+                'scope': 'profile email',
+                allow_field[client_type]: True,
             },
             follow=True,
         )
 
+        check_response = getattr(self, '_check_{}_response'.format(client_type))
+        check_response(response)
+
+    def _check_dot_response(self, response):
+        """
+        Check that django-oauth-toolkit gives an appropriate authorization response.
+        """
+        # django-oauth-toolkit tries to redirect to the user's redirect URL
+        self.assertEqual(response.status_code, 404)  # We used a non-existent redirect url.
+        expected_redirect_prefix = u'{}?'.format(DUMMY_REDIRECT_URL)
+        self._assert_startswith(self._redirect_destination(response), expected_redirect_prefix)
+
+    def _check_dop_response(self, response):
+        """
+        Check that django-oauth2-provider gives an appropriate authorization response.
+        """
+        # django-oauth-provider redirects to a confirmation page
         self.assertEqual(response.status_code, 200)
+        self.assertEqual(self._redirect_destination(response), u'http://testserver/oauth2/authorize/confirm')
 
-        # check form is in context and form params are valid
         context = response.context_data  # pylint: disable=no-member
-        self.assertIn('form', context)
-        self.assertIsNone(context['form']['authorize'].value())
+        form = context['form']
+        self.assertIsNone(form['authorize'].value())
 
-        self.assertIn('oauth_data', context)
         oauth_data = context['oauth_data']
         self.assertEqual(oauth_data['redirect_uri'], DUMMY_REDIRECT_URL)
         self.assertEqual(oauth_data['state'], 'random_state_string')
+        self.assertEqual(oauth_data['scope'], 6)
```

Comment:
Yeah.  I hadn't tracked it down, because handling scopes is outside the purview of this ticket.  I did want to capture that scope data was being handled here.  It looks like that number represents `provider.constants.READ_WRITE`, but I'm not sure why that scope got chosen.